### ISSUE TITLE
Kernel procedure table

### ISSUE THREAD

**Issue by Al-Kindi-0** [0xPolygonMiden/miden-vm#1247]
The kernel procedure (virtual) table keeps track of all unique kernel functions roots. The current implementation uses the address column of the kernel ROM chiplet to update this table. This, however, makes checking boundary constraints from the point of view of the verifier difficult if not impossible.
We should investigate how to change the updates to this table in order to fix the above.
Also, there is a potential discrepancy between the alphas used on the side of the decoder and the side of the kernel ROM chiplet and this should also be fixed.

**Comment by plafer** [comment]
> This, however, makes checking boundary constraints from the point of view of the verifier difficult if not impossible.

Could you expand on this? AFAIU, the boundary constraint for the verifier is to check that the first row of the virtual table column is equal to [this](https://github.com/0xPolygonMiden/miden-vm/blob/cdbfaed94de2d06976e7eac10f956186f2f6c8a5/processor/src/chiplets/aux_trace/mod.rs#L93-L102), which it can run since all the kernel procedure hashes are in the public inputs.

> Also, there is a potential discrepancy between the alphas used on the side of the decoder and the side of the kernel ROM chiplet and this should also be fixed.

How so? Would our tests fail if somehow we weren't consistent with the use of alphas?

**Comment by Al-Kindi-0** [comment]
> > This, however, makes checking boundary constraints from the point of view of the verifier difficult if not impossible.
> 
> Could you expand on this? AFAIU, the boundary constraint for the verifier is to check that the first row of the virtual table column is equal to [this](https://github.com/0xPolygonMiden/miden-vm/blob/cdbfaed94de2d06976e7eac10f956186f2f6c8a5/processor/src/chiplets/aux_trace/mod.rs#L93-L102), which it can run since all the kernel procedure hashes are in the public inputs.

Agreed, but note that on line 96 we include the address which will not be known to the verifier and this is what this issue is in principle about.

> 
> > Also, there is a potential discrepancy between the alphas used on the side of the decoder and the side of the kernel ROM chiplet and this should also be fixed.
> 
> How so? Would our tests fail if somehow we weren't consistent with the use of alphas?

Agreed, but note that this issue is older than the PR which fixed the relevant bus so maybe that was fixed there.

**Comment by Al-Kindi-0** [comment]
Basically, `addr` should be called index. Then I think this should not be a problem for the verifier.

**Comment by plafer** [comment]
closed by #1653
